itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/queryhistory/repository/TestIcebergRepository.java

```diff
@@ -0,0 +1,115 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.queryhistory.repository;
+
+import org.apache.hadoop.hive.ql.queryhistory.schema.QueryHistorySchemaTestUtils;
+import org.apache.iceberg.mr.mapred.Container;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.data.Record;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.ServiceContext;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.queryhistory.QueryHistoryService;
+import org.apache.hadoop.hive.ql.queryhistory.repository.IcebergRepositoryForTest;
+import org.apache.hadoop.hive.ql.queryhistory.schema.DummyQueryHistoryRecord;
+import org.apache.hadoop.hive.ql.queryhistory.schema.QueryHistoryRecord;
+import org.apache.hadoop.hive.ql.queryhistory.schema.IcebergQueryHistoryRecord;
+import org.apache.hadoop.hive.ql.queryhistory.schema.QueryHistorySchema;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapred.FileInputFormat;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.RecordReader;
+import org.apache.hadoop.mapred.Reporter;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class TestIcebergRepository {
+  private static final Logger LOG = LoggerFactory.getLogger(TestIcebergRepository.class);
+  private static final ServiceContext serviceContext = new ServiceContext(() -> DummyQueryHistoryRecord.SERVER_HOST,
+      () ->  DummyQueryHistoryRecord.SERVER_PORT);
+  private final Queue<QueryHistoryRecord> queryHistoryQueue = new LinkedBlockingQueue<>();
+
+  /*
+   * This unit test asserts that the created record is persisted as expected and the values made their way to the
+   * iceberg table.
+   */
+  @Test
+  public void testPersistRecord() throws Exception {
+    HiveConf conf = new HiveConf();
+
+    conf.setBoolVar(HiveConf.ConfVars.HIVE_CLI_TEZ_INITIALIZE_SESSION, false);
+    conf.setIntVar(HiveConf.ConfVars.HIVE_QUERY_HISTORY_PERSIST_MAX_BATCH_SIZE, 0);
+    conf.setVar(HiveConf.ConfVars.HIVE_QUERY_HISTORY_REPOSITORY_CLASS, IcebergRepositoryForTest.class.getName());
```

Comment:
please replace it with a production impl and set needed config ("iceberg.engine.hive.lock-enabled", "false")